src/npc.c

```diff
@@ -224,49 +806,261 @@ INCLUDE_ASM(s32, "npc", func_8003BA60);
+            func_8003CFA0(npc, idx);
+            break;
+    }
+    npc->decorationType[idx] = 0;
+}
+
+void func_8003C61C(Npc* npc, s32 idx) {
 
-INCLUDE_ASM(s32, "npc", func_8003C428);
+    switch (npc->decorationType[idx]) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+            break;
+        case 6:
+            func_8003CFA8(npc, idx);
+            break;
+    }
+}
 
-INCLUDE_ASM(s32, "npc", npc_update_decorations);
+void func_8003C658(Npc* npc, s32 idx) {
+}
+
+void func_8003C660(Npc* npc, s32 idx) {
+}
 
-INCLUDE_ASM(s32, "npc", func_8003C53C);
+void fx_sweat(s32, f32, f32, f32, f32, f32, s32);
 
-INCLUDE_ASM(s32, "npc", func_8003C61C);
+void func_80070A30(s32, f32, f32, f32, f32, s32, EffectInstanceData**);
+void func_80070EB0(s32, f32, f32, f32, f32, EffectInstanceData**);
 
-void func_8003C658(void) {
+void func_8003C668(Npc* npc, s32 idx) {
+    EffectInstanceData* instanceData;
+
+    switch (npc->decorationInitialised[idx]) {
+        case 0:
+            func_80070EB0(2, npc->pos.x, npc->pos.y, npc->pos.z, 1.0f, &npc->decorations[idx]);
+            npc->decorationInitialised[idx] = 1;
+            break;
+        case 1:
+            break;
+        default:
+            return;
+
+    }
+
+    instanceData = npc->decorations[idx]->data;
+    instanceData->pos.x = npc->pos.x;
+    instanceData->pos.y = npc->pos.y;
+    instanceData->pos.z = npc->pos.z;
+    instanceData->scale.x = (npc->scale.x * npc->collisionRadius) * 0.01;
+    instanceData->scale.y = (npc->scale.y * npc->collisionHeight) * 0.01;
+    instanceData->unk_64 = npc->renderYaw;
+}
+
+void func_8003C78C(Npc* npc, s32 idx) {
+    npc->decorations[idx]->data->unk_2C = 5;
+}
+
+void func_8003C7A8(Npc* npc, s32 idx) {
+    switch (npc->decorationInitialised[idx]) {
+        case 0:
+            if (npc->yawCamOffset > 90) {
+                fx_sweat(0, npc->pos.x, npc->pos.y + npc->collisionHeight, npc->pos.z, 5.0f, 45.0f, 0x14);
+            } else {
+                fx_sweat(0, npc->pos.x, npc->pos.y + npc->collisionHeight, npc->pos.z, 5.0f, -45.0f, 0x14);
+            }
+            npc->decorationUnk[idx] = 10;
+            npc->decorationInitialised[idx] = 1;
+            break;
+        case 1:
+            if (npc->decorationUnk[idx] != 0) {
+                npc->decorationUnk[idx]--;
+                return;
+            }
+            npc->decorationInitialised[idx] = 0;
+            break;
+    }
 }
 
-void func_8003C660(void) {
+void func_8003C8AC(Npc* npc, s32 idx) {
 }
 
-INCLUDE_ASM(s32, "npc", func_8003C668);
+void func_8003C8B4(Npc* npc, s32 idx) {
+    EffectInstanceData* instanceData;
 
-INCLUDE_ASM(s32, "npc", func_8003C78C);
+    switch (npc->decorationInitialised[idx]) {
+        case 0:
+            func_80070A30(0, npc->pos.x, npc->pos.y + npc->collisionHeight, npc->pos.z, 20.0f, 3, &npc->decorations[idx]);
+            npc->decorationInitialised[idx] = 1;
+            return;
```

Comment:
break instead of return? and below

Reply:
there's more to this function (setting the instanceData stuff) so just break is functionally different. We don't want to set the instanceData members if it was one of these specific cases.